### DISCUSSION TITLE
[Bug] How to create Custom Entry

### DISCUSSION THREAD

**Discussion by Shraddha996** [xamarin/Xamarin.Forms#11304] [General]
Hello,
As I am new to this platform
I want to create an Entry Box like this with header Field Name and I almost tried everything but unable to find any proper solution
![3](https://user-images.githubusercontent.com/39983941/86488178-c4e31700-bd7d-11ea-89ed-2131c6a30dbb.PNG)

Please Help!!!!!!!!

**Comment by davidortinau** [comment]
Hi @Shraddha996 you might be able to find something via NuGet/GitHub that already does this, but I would suggest first trying to make a composite control (commonly also called a user control). Basically start by adding a `ContentView` and giving it a name like CustomEntry. Then the root element I would make a Grid and do something like (structure only):

```xml
<Grid>
  <Frame>
    <Entry />
    <Label Text="EMAIL*" />
  </Frame>
</Grid>
```

For styling:
* The Grid let's me position things in a z-stack as I want. 
* The Frame can take care of the border and rounded corners, and the shadow (PancakeView is a powerful implementation of Frame you may want to grab that NuGet). 
* The Entry you'll want to grab a an effect to hide the borders on iOS and underline on Android native controls. [This project](https://github.com/muak/AiForms.Effects) seems like it might have something you can use.
* The label can be positioned over the Frame with a solid white background to give the panel-like look

Expose as a [bindable property](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/xaml/bindable-properties) the things you need, probably just the Entry Text.

Then in your ContentPage you just use it like any other custom control by adding the project namespace.

```xml
<ContentPage
  xmlns:custom="your namespace / assembly info">
  <StackLayout>
    <custom:CustomerEntry Text="{Binding Email}" />
  </StackLayout>
</ContentPage
```

This is all psuedo-code of course. I've done this several times... though I can't think of an Entry off the top of my head. Here are some custom controls in an older project: https://github.com/davidortinau/ProfileLayout/tree/master/ProfileLayout/Views

**Reply by davidortinau** [reply]
Ahhhh. Here's an Entry effect that removes the background and sets the border size on iOS:

https://github.com/davidortinau/HowYouSay/blob/master/HowYouSay.Shared/Views/VocabEntryDetailsPage.xaml#L41

https://github.com/davidortinau/HowYouSay/blob/master/iOS/Effects/InputViewEffect.cs

In case you're not familiar with using effects. https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/effects/

Again, if you can find the effects already out there, that's easier and you won't have to master UIKit and Android Widgets.
